packages/liveblocks-core/src/__tests__/room.test.ts

```diff
@@ -954,6 +955,45 @@ describe("room", () => {
     expect(room.history.canUndo()).toBe(false);
   });
 
+  test("missing storage keys are properly initialized using initialStorage, even when they are already part of the storage tree", async () => {
+    const initialPresence = {};
+    const initialStorage = {
+      list: new LiveList([13, 42]),
+      map: new LiveMap([["a", 1]]),
+      obj: new LiveObject({ x: 0 }),
+    };
+
+    // Create the same room instance twice (using the same, global,
+    // initialStorage instance)
+    for (let i = 0; i < 2; i++) {
+      const { room, wss } = createTestableRoom(
+        initialPresence,
+        undefined,
+        undefined,
+        undefined,
+        initialStorage
+      );
+      wss.onConnection((conn) => {
+        conn.server.send(
+          serverMessage({
+            type: ServerMsgCode.INITIAL_STORAGE_STATE,
+            items: [["root", { type: CrdtType.OBJECT, data: {} }]],
+            //                                              ^^
+            //                                   NOTE: Storage is initially
+            //                                   empty, so initialStorage keys
+            //                                   will get added
+          })
+        );
+      });
+      room.connect();
+      try {
+        await room.getStorage();
+      } finally {
+        room.destroy();
+      }
+    }
+  });
```

Comment:
This test fails on `main`, but passes with [this fix](https://github.com/liveblocks/liveblocks/pull/1245/files#diff-582bfabb50c03434ec253eb453124d78711e62f2afac7270fd5813ab638f90cf) applied.